public/app/plugins/panel/canvas/utils.ts

```diff
@@ -107,28 +106,155 @@ export function onAddItem(sel: SelectableValue<string>, rootLayer: FrameState |
   }
 }
 
-export function getDataLinks(ctx: DimensionContext, cfg: TextConfig, textData: string | undefined): LinkModel[] {
-  const panelData = ctx.getPanelData();
+/*
+ * Provided a given field add any matching data links
+ * Mutates the links object in place which is then returned by the `getDataLinks` function downstream
+ */
+const addDataLinkForField = (
+  field: Field<unknown>,
+  data: string | undefined,
+  linkLookup: Set<string>,
+  links: Array<LinkModel<Field>>
+): void => {
+  if (field?.getLinks) {
+    const disp = field.display ? field.display(data) : { text: `${data}`, numeric: +data! };
+    field.getLinks({ calculatedValue: disp }).forEach((link) => {
+      const key = `${link.title}/${link.href}`;
+      if (!linkLookup.has(key)) {
+        links.push(link);
+        linkLookup.add(key);
+      }
+    });
+  }
+};
+
+// TODO: This could be refactored a fair amount, ideally the element specific config code should be owned by each element and not in this shared util file
```

Comment:
mhm

was about to suggest DRYing all the copy/paste out, but that would make extracting each to separate elements more work maybe :thinking: